server/src/main/java/io/druid/server/emitter/HttpEmitterSSLClientConfig.java

```diff
@@ -24,15 +24,7 @@
 import io.druid.metadata.PasswordProvider;
 
 /**
- * This is kept separate from {@link io.druid.java.util.emitter.core.HttpEmitterConfig} because {@link PasswordProvider}
- * is currently located in druid-api. The java-util module which contains HttpEmitterConfig cannot import
- * PasswordProvider because this would introduce a circular dependence between java-util and druid-api.
- *
- * PasswordProvider could be moved to java-util, but PasswordProvider is annotated with
- * {@link io.druid.guice.annotations.ExtensionPoint}, which would also have to be moved.
- *
- * It would be easier to resolve these issues and merge the TLS-related config with HttpEmitterConfig once
- * https://github.com/druid-io/druid/issues/4312 is resolved, so the TLS config is kept separate for now.
+ * TODO merge with {@link io.druid.java.util.emitter.core.HttpEmitterConfig}
```

Comment:
can we fix this TODO?

Reply:
okay, can we add more context to what the TODO means and link it to an issue?

Reply:
I think it makes sense, in this PR, to change nothing about the content and just collapse the modules.

Reply:
@b-slim previous comment was a lengthy explanation why those config classes couldn't be merged at that time. Now they could. I didn't want to add explanations why it *should* be done, because previously it wasn't as well (it is considered obvious, I suppose.)

Anyway, in general we don't usually add TODOs and favor issues on github; I've created #5491.